locksmith/src/controllers/v2/ticketsController.ts

```diff
@@ -98,4 +106,36 @@ export class TicketsController {
       return response.sendStatus(500)
     }
   }
+
+  /**
+   * Function that serves a QR code.
+   * It can only be called by a lock manager (otherwise anyone can create a valid QR code that will be used to check-in!)
+   * @param request
+   * @param response
+   * @returns
+   */
+  async getQrCode(request: Request, response: Response) {
+    try {
+      const lockAddress = Normalizer.ethereumAddress(request.params.lockAddress)
+      const network = Number(request.params.network)
+      const tokenId = request.params.tokenId.toLowerCase()
+
+      const qrCode = (
+        await generateQrCode({
+          network,
+          lockAddress,
+          tokenId,
+        })
+      ).replace('data:image/gif;base64,', '')
+      const img = Buffer.from(qrCode, 'base64')
+
+      response.writeHead(200, {
+        'Content-Type': 'image/gif',
+      })
+      return response.end(img)
+    } catch (err) {
+      logger.error(err)
+      return response.sendStatus(500)
```

Comment:
A generic message which details what failed would be good here. "Failed to generate QR code"